geospaas_harvesting/ingesters.py

```diff
@@ -310,35 +319,58 @@ def _get_xml_namespace(self, root):
             self.LOGGER.warning('Could not find XML namespace while reading DDX metadata')
         return namespace_prefix
 
-    def _extract_global_attributes(self, root):
-        """Extracts the global attributes of a dataset from a DDX document"""
+    def _extract_attributes(self, root):
+        """
+        Extracts the global or specific attributes of a dataset or specific ones from a DDX document
+
+        "x_path_global" is pointing to the 'NC_GLOBAL' part of response of HTML to obtain general
+        information.
+        "x_path_specific" is adaptive part (or many parts) of response of HTML to obtain the
+        raw names of parameters for cumulative normalization by metanorm package as the core of the
+        harvesting processing.
+        """
         self.LOGGER.debug("Getting the dataset's global attributes.")
         namespaces = {'default': self._get_xml_namespace(root)}
-        global_attributes = {}
-        for attribute in root.findall(
-                f"./default:Attribute[@name='{self.GLOBAL_ATTRIBUTES_NAME}']/default:Attribute",
-                namespaces):
-
-            global_attributes[attribute.get('name')] = attribute.find(
+        extracted_attributes = {}
+        x_path_global = "./default:Attribute[@name='NC_GLOBAL']/default:Attribute"
+        x_path_specific = "./default:Grid/default:Attribute[@name='standard_name']"
+        # finding the global metadata
+        for attribute in root.findall(x_path_global, namespaces):
+            extracted_attributes[attribute.get('name')] = attribute.find(
                 "./default:value", namespaces).text
-
-        return global_attributes
+        # finding the parameters of the dataset that are declared in the online source (specific metadata)
```

Comment:
Please also crop comment lines to 100 characters

Reply:
Done in the next commit